PSL/Peripherals.py

```diff
@@ -39,12 +40,28 @@ class I2C():
     tg = 100
     MAX_SAMPLES = 10000
 
-    def __init__(self, H):
+    def __init__(self, H, address=None):
         self.H = H
         from PSL import sensorlist
         self.SENSORS = sensorlist.sensors
         self.buff = np.zeros(10000)
 
+        if address is not None:
+            self.start = partial(self.start, address)
```

Comment:
The partial methods are shadowing existing methods. This will change the method signatures depending on how the class is instantiated, which is not OK.

Reply:
@bessman thanks for your review.
Is it OK to create I2C_slave (sub class for I2C) which would be initialized with address?

Reply:
Yes, but if the subclass overrides methods from the parent class, the overriding methods should ideally have the same signature as the overridden methods. It is acceptable for the overriding methods to take _additional_ arguments, but not fewer.

Rather, consider splitting up the I2C class into I2CMaster and I2CSlave, where Master would handle slave-independent functionality (e.g. device scanning and bus configuration) and Slave would handle sending/receiving of data. This would break backward compatibility and require changes to all sensor classes which use I2C, but I don't see a way to make this change cleanly without breaking the existing API.

Reply:
Actually, there is a way to preserve the current API:
Write the I2CMaster and I2CSlave classes as mixins, and create a new I2C class which inherits from both. Then the I2C class, which is now subclass, could override the send/receive methods from I2CSlave to additionally require an address. New sensors could be written using the new classes directly, while already implemented sensors keep working as is.

Reply:
ok :+1: